packages/renderer/src/lib/preferences/PreferencesResourcesRenderingCopyButton.svelte

```diff
@@ -0,0 +1,36 @@
+<script lang="ts">
+import { onMount } from 'svelte';
+import Fa from 'svelte-fa';
+import { faPaste } from '@fortawesome/free-solid-svg-icons';
+import Tooltip from '../ui/Tooltip.svelte';
+
+let platform: string = '';
+export let path: string;
+
+onMount(async () => {
+  platform = await window.getOsPlatform();
+});
+
+function copyResourceSocketAddrToClipboard() {
+  let prefix = '';
+  if (platform === 'win32') {
+    prefix = 'npipe';
+  } else {
+    prefix = 'unix';
+  }
+
+  const url = `${prefix}://${path}`;
```

Comment:
you could apply this transformation in the onMount

like

```ts
let url: string = '';
onMount(....

url = ..
})

```


and in the code

{#if url}
   <Tooltip ....>

{/if}

Reply:
Good point. The path will only be given when the component we create the component, so it feels right to have inside ```onMount``` and the click only does the copy.

Reply:
I developed the test but, it needs 3 ticks to update have the DOM updated. I'm not yet familiar with this test framework and s, can you spot what I'm missing?